tools/topology/topology2/include/components/micsel.conf

```diff
@@ -0,0 +1,101 @@
+#
+# Common widget micsel
+#
+# A generic micsel widget. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.micsel.N.attribute_name"
+#
+# Usage: this component can be used by instantiating it in the parent object. i.e.
+#
+# 	Object.Widget.micsel."N" {
+#
+#	}
+#
+# Where N is the unique instance number for the micsel object within the same alsaconf node.
+
+Class.Widget."micsel" {
+	#
+	# Pipeline ID for the micsel object
+	#
+	DefineAttribute."index" {}
+
+	#
+	# Micsel object instance
+	#
+	DefineAttribute."instance" {}
+
+	DefineAttribute."cpc" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	DefineAttribute."num_audio_formats" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	#include common component definition
+	<include/components/widget-common.conf>
+
+	#
+	# micsel component UUID
+	#
+	DefineAttribute."uuid" {
+		type "string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.uuid"
+	}
+
+	DefineAttribute."process_type" {
+		type "string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_process.string"
+	}
+
+	DefineAttribute."payload_with_output_fmt" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.bool"
+	}
+
+	attributes {
+		#
+		# The micsel widget name would be constructed using the index and
+		# instance attributes. For example: "micsel.0.1".
+		#
+		!constructor [
+			"index"
+			"instance"
+		]
+
+		#
+		# mandatory attributes that must be provided when the class is instantiated
+		#
+		!mandatory [
+			"no_pm"
+			"uuid"
+		]
+
+		#
+		# immutable attributes cannot be modified in the object instance
+		#
+		!immutable [
+			"uuid"
+		]
+
+		unique	"instance"
+	}
+
+	#
+	# Default attributes for micsel
+	#
+	type		"effect"
+	payload_with_output_fmt		1
+
+	num_audio_formats 1
+	#UUID: 32FE92C1-1E17-4FC2-9758-C7F3542E980A
+	uuid 		"c1:92:fe:32:17:1e:c2:4f:97:58:c7:f3:54:2e:98:0a"
+	no_pm 		"true"
+	core_id	0
+	cpc 5000
```

Comment:
how is this determined?

Reply:
@libinyang when you give someone an answer like this, you should know what the next question is going to be right? Please go back to them and find out how this was derived and add a comment. Thanks!

Reply:
cps = cpc * sampling frequency // sample group in ms * frame size, which can be found in the platform config file. So you can calculate the cpc value.
ps: I won't add the cpc calculation description in the patch

Reply:
> sorry what is cps?

I think CPS is "Cycles Per Second" after grep the sof fw src code.

Cited more from CoE team 
"frame_size  - number of milliseconds that one frame consists of.
 cpc - number of cycles per chunk, chunk is a data processed during one frame. "

And cps value can be found in the file platform config file in the rimage.